### DISCUSSION TITLE
Including Local or Private Git Packages in Flet Build

### DISCUSSION THREAD

**Discussion by amcraig** [flet-dev/flet#4546] [Q&A] [answered]
### Question

Hi all, 
I've tried including my python package(not on PyPi) through both relative paths to the whl/tar.gz and via git in both `requirements.txt` and in `pyproject.toml` (including poetry) but any attempts I do fail due to a `distutils Module not found` error or `KeyError: 'version'`.

Does anyone have a guaranteed way to provide a local/private python package to Flet in the build process?
Thanks!

### Code sample

```python
##### Pyproject Poetry 
[tool.poetry]
name = "file_tracker"
version = "0.5.0"
description = "redacted"
authors = ["amcraig"]

[tool.poetry.dependencies]
python = "^3.10"
private_package = { git = "https://github.com/private/package.git" }
flet = "^0.25.1"

##### requirements.txt
python==3.10
flet
datateam @ git+https://github.com/private/package
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by FeodorFitsner** [comment]
Right, parsing non-version dependencies from Poetry-style dependencies is not yet implemented. It should work with `requirements.txt` though. If both `requirements.txt` and `pyproject.toml` exist then `requirements.txt` will be ignored. I'm adding the issue to implement that.

**Reply by FeodorFitsner** [reply, marked as answer]
Subscribe to https://github.com/flet-dev/flet/issues/4547
